File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left=0
        right=len(heights)-1
        maximum=0
        while left<right:
            s=(right-left)*min(heights[right],heights[left])
            if(heights[right]>heights[left]):
                left+=1
            else:
                right-=1
            maximum=max(s,maximum)
        return maximum
        

File: Data Structures & Algorithms/balanced-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
private:
    int valid(TreeNode* root,bool &val)
    {
        if(!root)   return 0;
        int l=valid(root->left,val);
        int r=valid(root->right,val);
        if(abs(l-r)>1)  val=0;
        return 1+max(l,r);

    }
public:
    bool isBalanced(TreeNode* root) {
        bool val=1;
        valid(root,val);
        return val;

        
    }
};
